packages/utils/src/instrument/fetch.ts

```diff
@@ -58,6 +59,15 @@ function instrumentFetch(onFetchResolved?: (response: Response) => void, skipNat
         startTimestamp: timestampInSeconds() * 1000,
       };
 
+      const body = parseFetchPayload(args);
```

Comment:
The main work that IMHO is left is to remove this code here, and to instead do this in the integration itself. Through the hint you should have access to the args, so you should be able to get the payload in there directly (in a hook) instead of doing this generically for all requests.

Then, you can inline the `getGraphQLRequestPayload` code too into the integration itself, which should make typing etc. stuff much easier. Ideally, this file is not touched at all, and the only thing that is added outside of the integration is the hooks configuration - everything else (more or less) is encapsulated in the integration :)